### DISCUSSION TITLE
InMemoryCosmosRepository How can i clear/dispose

### DISCUSSION THREAD

**Discussion by AronRidgway-Castellan** [IEvangelist/azure-cosmos-dotnet-repository#206] [Q&A]
Hi, I using the InMemoryCosmosRepository for integration tests, and need to clear after each test. Is there an easy easy way to do this? I've tried disposing of the IserviceCollection but it still has existing data in it?

Thanks
Aron

**Comment by mumby0168** [comment]
> Hi, I using the InMemoryCosmosRepository for integration tests, and need to clear after each test. Is there an easy easy way to do this? I've tried disposing of the IserviceCollection but it still has existing data in it?
> 
> Thanks Aron

Hi, the way we tend to use this is we create a brand new `ServiceCollection` for each test, see this example see here -> https://github.com/mumby0168/Nomad.Sorter/blob/main/tests/Nomad.Sorter.Unit.Tests/Infrastructure/Comsos/Repositories/ParcelRepositoryTests.cs

**Reply by AronRidgway-Castellan** [reply]
Thank you, i will give it a try.

**Reply by mumby0168** [reply]
Any luck with this one?

**Reply by IEvangelist** [reply]
We could probably add `Reset` functionality to the `InMemoryRepository` impl
